AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1854,10 +1884,18 @@ - (void)_removeSubnode:(ASDisplayNode *)subnode
   [subnode __setSupernode:nil];
 }
 
-// NOTE: You must not called this method while holding the receiver's property lock. This may cause deadlocks.
 - (void)removeFromSupernode
 {
+  ASDisplayNodeAssert(self.supernode.automaticallyManagesSubnodes == NO, @"Attempt to manually remove subnode from node with automaticallyManagesSubnodes=YES. Node: %@", self);
+    
+  [self _removeFromSupernode];
```

Comment:
This is unfortunate. If you don't really care how the supernode is implemented (which is usually important), this will be very irritating.